File: app/src/main/java/dev/ragnarok/fenrir/db/model/entity/feedback/ReplyCommentEntity.java
package dev.ragnarok.fenrir.db.model.entity.feedback;

import com.google.gson.annotations.SerializedName;

import dev.ragnarok.fenrir.db.model.entity.CommentEntity;
import dev.ragnarok.fenrir.db.model.entity.Entity;
import dev.ragnarok.fenrir.db.model.entity.EntityWrapper;
import dev.ragnarok.fenrir.model.feedback.FeedbackType;

public class ReplyCommentEntity extends FeedbackEntity {

    @SerializedName("commented")
    private EntityWrapper commented = new EntityWrapper();
    @SerializedName("ownComment")
    private CommentEntity ownComment;
    @SerializedName("feedbackComment")
    private CommentEntity feedbackComment;

    @SuppressWarnings("unused")
    public ReplyCommentEntity() {
    }

    public ReplyCommentEntity(@FeedbackType int type) {
        setType(type);
    }

    public Entity getCommented() {
        return commented.get();
    }

    public ReplyCommentEntity setCommented(Entity commented) {
        this.commented = new EntityWrapper().wrap(commented);
        return this;
    }

    public CommentEntity getFeedbackComment() {
        return feedbackComment;
    }

    public ReplyCommentEntity setFeedbackComment(CommentEntity feedbackComment) {
        this.feedbackComment = feedbackComment;
        return this;
    }

    public CommentEntity getOwnComment() {
        return ownComment;
    }

    public ReplyCommentEntity setOwnComment(CommentEntity ownComment) {
        this.ownComment = ownComment;
        return this;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/db/model/entity/feedback/ReplyCommentEntity.kt
package dev.ragnarok.fenrir.db.model.entity.feedback

import androidx.annotation.Keep
import dev.ragnarok.fenrir.db.model.entity.CommentEntity
import dev.ragnarok.fenrir.db.model.entity.DboEntity
import dev.ragnarok.fenrir.model.feedback.FeedbackType
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Keep
@Serializable
@SerialName("reply_comment")
class ReplyCommentEntity : FeedbackEntity {
    private var commented: DboEntity? = null

    var ownComment: CommentEntity? = null
        private set

    var feedbackComment: CommentEntity? = null
        private set

    @Suppress("UNUSED")
    constructor()
    constructor(@FeedbackType type: Int) {
        this.type = type
    }

    fun getCommented(): DboEntity? {
        return commented
    }

    fun setCommented(commented: DboEntity?): ReplyCommentEntity {
        this.commented = commented
        return this
    }

    fun setFeedbackComment(feedbackComment: CommentEntity?): ReplyCommentEntity {
        this.feedbackComment = feedbackComment
        return this
    }

    fun setOwnComment(ownComment: CommentEntity?): ReplyCommentEntity {
        this.ownComment = ownComment
        return this
    }
}